platform/darwin/test/MGLStyleValueTests.swift

```diff
@@ -1,8 +1,61 @@
 import XCTest
 import Mapbox
 
-
+#if os(iOS) || os(watchOS) || os(tvOS)
+typealias MGLColor = UIColor
+#elseif os(macOS)
+typealias MGLColor = NSColor
+#endif
+    
 extension MGLStyleValueTests {
+    func assertColorsEqualWithAccuracy(_ actual: MGLColor, _ expected: MGLColor, accuracy: Float = 1/255) {
+        var actualComponents : [CGFloat] = [0, 0, 0, 0]
+        var expectedComponents : [CGFloat] = [0, 0, 0, 0]
+        actual.getRed(&(actualComponents[0]), green: &(actualComponents[1]), blue: &(actualComponents[2]), alpha: &(actualComponents[3]))
+        expected.getRed(&(expectedComponents[0]), green: &(expectedComponents[1]), blue: &(expectedComponents[2]), alpha: &(expectedComponents[3]))
+        for (ac, ec) in zip(actualComponents, expectedComponents) {
+            XCTAssertEqualWithAccuracy(Float(ac), Float(ec), accuracy: accuracy)
+        }
+    }
+    
+    func assertColorValuesEqual(_ actual: MGLStyleValue<MGLColor>, _ expected: MGLStyleValue<MGLColor>) {
+        if let actualConstant = actual as? MGLStyleConstantValue<MGLColor> {
+            XCTAssertTrue(expected is MGLStyleConstantValue<MGLColor>)
+            assertColorsEqualWithAccuracy(actualConstant.rawValue, (expected as! MGLStyleConstantValue<MGLColor>).rawValue)
+        } else if let actualFunction = actual as? MGLStyleFunction<MGLColor> {
+            XCTAssertTrue(expected is MGLStyleFunction<MGLColor>)
+            let expectedFunction = expected as! MGLStyleFunction<MGLColor>
+            XCTAssertEqual(actualFunction.interpolationBase, expectedFunction.interpolationBase)
+            XCTAssertEqual(actualFunction.interpolationMode, expectedFunction.interpolationMode)
+            // TODO: assert default values are equal
+            guard let actualStops = actualFunction.stops else {
+                XCTAssertNil(expectedFunction.stops)
+                return
+            }
+            let expectedStops = expectedFunction.stops!
+            XCTAssertEqual(actualStops.keys.count, expectedStops.keys.count)
+            let actualKeys = actualStops.keys.sorted(by: { String(describing: $0) < String(describing: $1) })
```

Comment:
Yeah -- I originally tried just using `<` as the sort function, but it seemed to think that the keys were of type `Any`.  I landed at the same conclusion as you: that the weird alphanumeric sort wouldn't matter here... _but_, alas, I think we should probably figure this key type thing out in more detail -- if the conversion stuff is turning numeric stop values into string values, that's a bug, and comparing stringified values of the keys like we're doing here would occlude it.